projects/plugins/boost/app/modules/image-size-analysis/Image_Size_Analysis_Fixer.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Automattic\Jetpack_Boost\Modules\Image_Size_Analysis;
+
+class Image_Size_Analysis_Fixer {
+
+	public static function setup() {
+		add_filter( 'the_content', array( __CLASS__, 'fix_content' ), 99 );
+		add_filter( 'wp_calculate_image_srcset', array( __CLASS__, 'fix_image_attachments' ), 10, 5 );
+		add_filter( 'wp_calculate_image_sizes', array( __CLASS__, 'fix_image_sizes' ), 10, 5 );
+	}
+
+	public static function fix_url( $url ) {
+		$url = preg_replace( '/-\d+x\d+\.jpg/', '.jpg', $url ); // remove "-1024x768" from the end of the URL.
+		$url = preg_replace( '/\?.*/', '', $url ); // remove query string.
+		$url = str_replace( 'i0.wp.com/', '', $url ); // remove i0.wp.com/ from the beginning of the URL.
+		return $url;
+	}
+
+	public static function get_fixes( $post_id ) {
+		static $fixes = array();
+
+		if ( isset( $fixes[ $post_id ] ) ) {
+			return $fixes[ $post_id ];
+		}
+
+		$fixes[ $post_id ] = get_post_meta( $post_id, 'jb_image_fixes', true );
+		if ( ! $fixes[ $post_id ] ) {
+			$fixes[ $post_id ] = array();
+		}
+
+		return $fixes[ $post_id ];
+	}
+
+	// phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+	public static function fix_image_attachments( $sources, $size_array, $image_src, $image_meta, $attachment_id ) {
+		global $post;
+
+		$fixes = self::get_fixes( $post->ID );
+
+		$image_width = 0;
+
+		// remove XxY dimension from $image_src as that's what's recorded by Image_Size_Analysis.
+		$image_src_key = self::fix_url( $image_src );
```

Comment:
Maybe `fix_url` should be called something that describes the return of the function? I'm not entirely clear currently what it returns, but in case what's left of the cleaned URL is just the domain (for example) then I'd call this `get_url_domain()` or along those lines.

Reply:
I had called it clean_up_url previously, as it removes things like GET parameters, and "i0.wp.com/". I guess it's normalis(z)ing the URL.

Reply:
Hmm, a phpDoc with examples would be nice to have there then :) - I don't think there's much difference between fix and normalize.